backend/alert.go

```diff
@@ -0,0 +1,32 @@
+package backend
+
+const (
+	CLOSED_INTERVAL RangeIntervalType = iota
+	LEFT_CLOSED_INTERVAL
+	RIGHT_CLOSED_INTERVAL
+)
+
+const (
+	PENDING AlertState = iota
+	FIRING
+	RESOLVED
+)
+
+// Denotes the range interval type i.e closed interval are intervals of the form [from, to] whereas
+// open intervals are intervals of the form [-inf, to] (RIGHT_CLOSED_INTERVAL) or [from, inf] (LEFT_CLOSED_INTERVAL).
+type RangeIntervalType int
+
+// denotes the state of the active alert
+type AlertState int
+
+type Range struct {
+	from float64
+	to   float64
+	t    RangeIntervalType
+}
+
+type Alert struct {
+	r Range
+	id int64
```

Comment:
the id can be a string with meaningful value, e.g. "filesystem full", "cpu idle", ...